src/main/src/main/java/org/geoserver/data/util/TemporalUtils.java

```diff
@@ -53,4 +57,37 @@ public static boolean isDateTimeFormatEnabled() {
         Object hint = Hints.getSystemDefault(Hints.DATE_TIME_FORMAT_HANDLING);
         return !Boolean.FALSE.equals(hint);
     }
+
+    /**
+     * Serialize date to format specified
+     *
+     * @param date date to be converted
+     * @param dateFormatPattern Pattern of dateFormat
+     * @return converted dateTime text
+     * @throws RuntimeException on conversion error
+     */
+    public static String serializeDateTime(Date date, String dateFormatPattern) {
+        try {
+            Object hint = Hints.getSystemDefault(Hints.LOCAL_DATE_TIME_HANDLING);
+
+            Calendar cal = toCalendar(date);
+            // if it's only a date, no time involved
+            if (date instanceof java.sql.Date) {
+                SimpleDateFormat formatter = new SimpleDateFormat(dateFormatPattern);
+                return formatter.format(date);
+            } else {
+                // timestamp handling
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateFormatPattern);
+                ZonedDateTime zonedDateTime;
+                if (Boolean.TRUE.equals(hint)) {
+                    zonedDateTime = cal.toInstant().atZone(ZoneId.systemDefault());
+                } else {
+                    zonedDateTime = cal.toInstant().atZone(TimeZone.getTimeZone("GMT").toZoneId());
```

Comment:
Possibly:

```suggestion
                    zonedDateTime = cal.toInstant().atZone(TimeOffset.UTC);
```

(then you may not need the generic `Exception`).

Reply:
seems more like a bug in the GeoJSON writer than a lack in the CSV writer to me